### ISSUE TITLE
UNIQUE constraint failure when RR deletes chapters

### ISSUE THREAD

**Issue by mattheys** [DanCardin/chapter-sync#8]
Hi, I think I'm running into this issue too https://github.com/DanCardin/chapter-sync/issues/4  

Mark of the Fool has recently deleted 100 chapters in preparation to publish his next book and it seems like the SQL Alchemy is trying to insert an Id that already exists.

So here it's trying to insert it as Series 1 number 209, however I already have up to 309 in the database from the chapters that were downloaded before it was deleted.

https://github.com/DanCardin/chapter-sync/blob/8a4f418e6e652461833dbf6ab4cdad2614af9754/src/chapter_sync/migrations/versions/63139caea00e_initial_tables.py#L62-L64

```
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: chapter.series_id, chapter.number
[SQL: INSERT INTO chapter (series_id, title, url, number, content, ebook, sent_at, published_at, created_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: (1, 'Chapter 827: Mastering Flesh and Metal', 'https://www.royalroad.com/fiction/41618/mark-of-the-fool/chapter/1636529/chapter-827-mastering-flesh-and-metal', 209, '<div class="chapter-inner chapter-content">\n                <p class="cnM4ZWYyMjU3ZGI2MzRiZTJiZDE1YTM2NzQ1ZjJlN2Y4">“Good job, Cedric!” Alex clapped ... (18094 characters truncated) ... <p class="cnM0NWI4MmUxZmI2ZTQ4NzY5NGYzNDcwMjgyYzBhMGZl">“Cedric’s going to go a little crazy over that one,” Drestra grinned.</p>\n            </div>', None, None, '2024-05-14 22:38:51.000000', '2024-06-05 12:37:30.084515')]
```

**Comment by DanCardin** [comment]
I have an in progress branch in which i attempt to deal with this, but likely @SparxySys 's solution is fine for the common case, in that it's based on what already exists rather than the mutable source... I'll probably just merge that then, so it's not just broken in the interim

**Comment by DanCardin** [comment]
merged and released v0.3.5. lmk if that addresses this issue for you, or if we need a more complex variant

**Comment by mattheys** [comment]
Yep, it's all up to date now.  Chapter order seems fine as well.  Thank you
